treeshr/TreeAddNode.c

```diff
@@ -160,6 +160,7 @@ int _TreeAddNode(void *dbid, char const *name, int *nid_out, char usage)
 #pragma GCC diagnostic pop
           for (i = strlen(node_name); i < sizeof(new_ptr->name); i++)
             new_ptr->name[i] = 0;
+          new_ptr->name[MAX_NAME_LEN] = 0;  // null terminate truncated names
```

Comment:
When adding a null terminator to a string we prefer to use '\0'  instead, helps convey the intent. If you add that, you can drop the comment, it will be conveyed by the code

Reply:
Agreed.